corehq/apps/integration/kyc/models.py

```diff
@@ -116,7 +125,115 @@ def get_user_objects_by_ids(self, obj_ids):
             UserDataStore.CUSTOM_USER_DATA,
             UserDataStore.USER_CASE,
         ):
-            return [CommCareUser.get_by_user_id(id_) for id_ in obj_ids]
+            return KycUser.from_hq_user_or_case_objects(
+                self,
+                [CommCareUser.get_by_user_id(id_) for id_ in obj_ids]
+            )
         elif self.user_data_store == UserDataStore.OTHER_CASE_TYPE:
             assert self.other_case_type
-            return CommCareCase.objects.get_cases(obj_ids, self.domain)
+            return KycUser.from_hq_user_or_case_objects(self, CommCareCase.objects.get_cases(obj_ids, self.domain))
+
+
+class KycUser:
+
+    def __init__(self, kyc_config, user_or_case_obj):
+        """
+        :param kyc_config: kyc configuration for the domain
+        :param user_or_case_obj: can be an instance of 'CommcareUser' or 'CommcareCase' based on the configuration.
+        """
+        self.kyc_config = kyc_config
+        self.user_or_case_obj = user_or_case_obj
+
+    @cached_property
+    def user_id(self):
+        if isinstance(self.user_or_case_obj, CommCareUser):
+            return self.user_or_case_obj.user_id
+        return self.user_or_case_obj.case_id
```

Comment:
`self.user_or_case_obj.user_id` and `self.user_or_case_obj.case_id` are not expensive to get, so I don't think that lazy-loading and caching them are improving performance. You could simplify the class by just setting `self.user_id` in `__init__()`:

```suggestion
    def __init__(self, kyc_config, user_or_case_obj):
        """
        :param kyc_config: kyc configuration for the domain
        :param user_or_case_obj: can be an instance of 'CommcareUser' or 'CommcareCase' based on the configuration.
        """
        self.kyc_config = kyc_config
        self.user_or_case_obj = user_or_case_obj
        if isinstance(user_or_case_obj, CommCareUser):
            self.user_id = user_or_case_obj.user_id
        else:
            self.user_id = user_or_case_obj.case_id
```

Reply:
Addressed in 0adcf65aa4ac963fc4680c338bca3c0fdbc57ecc